agent/agent.go

```diff
@@ -633,6 +634,7 @@ func (a *Agent) runTask(ctx context.Context, tcInput *taskContext, nt *apimodels
 	}
 
 	defer a.killProcs(ctx, tc, false, "task is finished")
+	defer a.clearGitConfig(tc, "task is finished")
```

Comment:
Does it make sense to clear this state at the end of the task group only, rather than after each task in the task group? Tasks groups generally preserve state across the entirety of the task group, so this behavior is inconsistent with task directory cleanup (which only happens at the end of the task group), and process cleanup (which has an option to keep processes across tasks in the same task group).